starlite/_signature/models/attrs_signature_model.py

```diff
@@ -0,0 +1,343 @@
+from __future__ import annotations
+
+import re
+import traceback
+from dataclasses import asdict
+from datetime import date, datetime, time, timedelta
+from functools import lru_cache, partial
+from pathlib import PurePath
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    cast,
+)
+from uuid import UUID
+
+from _decimal import Decimal
+from dateutil.parser import parse
+from pytimeparse.timeparse import timeparse
+from typing_extensions import get_args
+
+from starlite._signature.field import SignatureField
+from starlite._signature.models.base import ErrorMessage, SignatureModel
+from starlite.connection import ASGIConnection, Request, WebSocket
+from starlite.datastructures import ImmutableState, MultiDict, State, UploadFile
+from starlite.exceptions import MissingDependencyException
+from starlite.params import BodyKwarg, DependencyKwarg, ParameterKwarg
+from starlite.types import Empty
+from starlite.utils import compact
+from starlite.utils.predicates import is_optional_union, is_union
+from starlite.utils.typing import get_origin_or_inner_type, make_non_optional_union, unwrap_union
+
+try:
+    import attr
+    import attrs
+    import cattrs
+except ImportError as e:
+    raise MissingDependencyException("attrs is not installed") from e
+
+if TYPE_CHECKING:
+    from starlite._signature.parsing import ParsedSignatureParameter
+    from starlite.plugins import PluginMapping
+
+key_re = re.compile("@ attribute (.*)|'(.*)'")
+
+__all__ = ("AttrsSignatureModel",)
+
+try:
+    import pydantic
+
+    def _structure_base_model(value: Any, cls: type[pydantic.BaseModel]) -> pydantic.BaseModel:
+        if isinstance(value, pydantic.BaseModel):
+            return value
+        return cls(**value)
+
+    pydantic_hooks: list[tuple[type[Any], Callable[[Any, type[Any]], Any]]] = [
+        (pydantic.BaseModel, _structure_base_model),
+    ]
+except ImportError:
+    pydantic_hooks = []
+
+
+def _pass_through_structure_hook(value: Any, _: type[Any]) -> Any:
+    return value
+
+
+def _pass_through_unstructure_hook(value: Any) -> Any:
```

Comment:
```suggestion
def _pass_through_unstructured_hook(value: Any) -> Any:
```

Reply:
`unstructure` is specific cattrs terminology, e.g. see: https://github.com/python-attrs/cattrs/blob/b5de29d4de2064246365c6d9d579ce3eabc3d01e/src/cattrs/converters.py#L229

Reply:
@peterschutt ty, confusing